### PR TITLE
Adding NuGet Tool Installer task

### PR DESCRIPTION
**Pull request by emanuelquintero**

### CODE REVIEW

File: Tasks/NuGetToolInstaller/nugetToolInstaller.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
+//import toolLib = require('vsts-task-tool-lib/tool');
```
Comment (pspill) [inline Tasks/NuGetToolInstaller/nugetToolInstaller.ts]: remove commented code

File: Tasks/NuGetToolInstaller/nugetToolInstaller.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+//import taskLib = require('vsts-task-lib/task');
+import {IExecOptions} from "vsts-task-lib/toolrunner";
+
+import * as toolLib from 'vsts-task-tool-lib/tool';
+import * as taskLib from 'vsts-task-lib/task';
+import * as restm from 'typed-rest-client/RestClient';
+import * as os from 'os';
+import * as path from 'path';
```
Comment (pspill) [inline Tasks/NuGetToolInstaller/nugetToolInstaller.ts]: path is not used

File: tasks/NuGetToolInstaller/test.ps1 [hunk 3]
```diff
@@ -0,0 +1,37 @@
+
```
Comment (ericsciple) [inline tasks/NuGetToolInstaller/test.ps1]: it looks like you may want to delete this before merging

File: Tasks/NuGetToolInstaller/task.json [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    "demands": [],
+    "instanceNameFormat": "Use NuGet $(versionSpec)",
+    "inputs": [
+        {
+            "name": "versionSpec",
+            "type": "string",
+            "label": "Version Spec",
+            "defaultValue": "3.5.x",
+            "required": true,
+            "helpMarkDown": "Version Spec of version to get.  Examples: 6.x, 4.x, 6.10.0, >=6.10.0"
```
Comment (ericsciple) [inline Tasks/NuGetToolInstaller/task.json]: you may want to update these examples

File: Tasks/NuGetToolInstaller/task.json [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+            "type": "boolean",
+            "label": "Stable Only",
+            "defaultValue": "false",
+            "required": false,
+            "helpMarkDown": "Specifies whether the version resolution will filter to stable only versions."
+        }
+    ],
+    "execution": {
+        "Node": {
+            "target": "nugetToolInstaller.js",
```
Comment (ericsciple) [inline Tasks/NuGetToolInstaller/task.json]: i believe one of the common tests will complain if you have upper chars in your js file

File: Tasks/NuGetToolInstaller/make.json [hunk 6]
```diff
@@ -0,0 +1,5 @@
+{
+    "taskResources": [
```
Comment (ericsciple) [inline Tasks/NuGetToolInstaller/make.json]: this file isn't needed

File: Tasks/NuGetToolInstaller/nugetToolInstaller.ts [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+    url: string
+}
+
+async function getNuGet(versionSpec: string, stableOnly: boolean) {
+    toolLib.debug('Trying to get tool from local cache');
+    let localVersions: string[] = toolLib.findLocalToolVersions('NuGet');
+    let version: string = toolLib.evaluateVersions(localVersions, versionSpec);
+
+    if (version) {
+        console.log('Resolved from tool cache: %s', version);
```
Comment (ericsciple) [inline Tasks/NuGetToolInstaller/nugetToolInstaller.ts]: May want to add TODO comments to loc. Also, I'm not sure, but the tool lib may be able to handle printing this type of info. I need to look into it.

**Comment by ericsciple** [comment]
i had a small comments only. looks good

File: Tasks/NuGetToolInstaller/package.json [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+    "NuGet"
+  ],
+  "author": "Microsoft Corporation",
+  "license": "MIT",
+  "bugs": {
+    "url": "https://github.com/microsoft/vsts-tasks/issues"
+  },
+  "homepage": "https://github.com/microsoft/vsts-tasks#readme",
+  "dependencies": {
+    "vsts-task-tool-lib": "0.3.1"
```
Comment (pspill) [inline Tasks/NuGetToolInstaller/package.json]: "^0.3.1" so we continue to get minor version updates

File: Tasks/NuGetToolInstaller/nugetToolInstaller.ts [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+    }
+    else {
+        toolLib.debug('Querying versions list');
+        let versionsUrl = 'https://dist.nuget.org/index.json';
+        let rest: restm.RestClient = new restm.RestClient('tool-sample');
```
Comment (pspill) [inline Tasks/NuGetToolInstaller/nugetToolInstaller.ts]: This is the user-agent string to use -- maybe "vsts-tasks/NuGetToolInstaller"

**Comment by emanuelquintero** [comment]
@pspill , @ericsciple I've addressed all of your comments.

**Pull request merged**